src/plugins/intel_cpu/src/nodes/gather.cpp

```diff
@@ -135,10 +139,16 @@ void Gather::initSupportedPrimitiveDescriptors() {
 
     // Implementation desc type will be redefined in the fn prepareParams if a kernel will be created.
     ov::element::Type dataPrecision = getOriginalInputPrecisionAtPort(GATHER_DATA);
+
+    canOptimizeCompressedEmbedding = false;
+    if ((decompressionSubtractPtr != nullptr) && (decompressionMultiplyPtr != nullptr)) {
```

Comment:
In fact to avoid strong cohesion with transformation logic we also need to check required conditions here: 2D tesnor, correct axis etc.

Reply:
Add stricter check. data precision, axis, data rank